.ci/pitest.groovy

```diff
@@ -0,0 +1,459 @@
+            case "mutatedClass":
+                mutatedClass = childNodeText
+                mutationClassPackage = mutatedClass.split("[A-Z]")[0]
+                break
+            case "mutatedMethod":
+                mutatedMethod = childNodeText
+                break
+            case "mutator":
+                mutator = childNodeText
+                break
+            case "description":
+                description = childNodeText
+                break
+            case "lineNumber":
+                lineNumber = Integer.parseInt(childNodeText)
+                break
+            case "lineContent":
+                lineContent = childNodeText
+                break
+        }
+    }
+    if (lineContent == null) {
+        final String mutationFileName = mutationClassPackage + sourceFile
+        final String startingPath = ".${SEPARATOR}src${SEPARATOR}main${SEPARATOR}java${SEPARATOR}"
+        final String javaExtension = ".java"
+        final String mutationFilePath = startingPath + mutationFileName
+                .substring(0, mutationFileName.length() - javaExtension.length())
+                .replaceAll("\\.", SEPARATOR) + javaExtension
+
+        final File file = new File(mutationFilePath)
+        lineContent = XmlUtil.escapeXml(file.readLines().get(lineNumber - 1).trim())
+    }
+    if (lineNumber == 0) {
+        lineNumber = -1
+    }
+    boolean unstable = false
+    final String unstableMutation = mutationNode.attribute("unstable")
+    if (unstableMutation != null) {
+        unstable = Boolean.parseBoolean(unstableMutation)
+    }
+    return new Mutation(sourceFile, mutatedClass, mutatedMethod, mutator, description,
+            lineContent, lineNumber, unstable)
+}
+
+/**
+ * Compare survived and suppressed mutations. The comparison passes successfully (i.e. returns 0)
+ * when:
+ * <ol>
+ *   <li>Survived and suppressed mutations are equal.</li>
+ *   <li>There are extra suppressed mutations but they are unstable
+ *     i.e. {@code unstable="true"}.</li>
+ * </ol>
+ * The comparison fails (i.e. returns 1) when:
+ * <ol>
+ *     <li>Survived mutations are not present in the suppressed list.</li>
+ *     <li>There are mutations in the suppression list that are not there is survived list.</li>
+ * </ol>
+ *
+ * @param survivedMutations A set of survived mutations
+ * @param suppressedMutations A set of suppressed mutations
+ * @param flag command line argument flag to determine output format
+ * @return {@code 0} if comparison passes successfully
+ */
+private static int compareMutations(Set<Mutation> survivedMutations,
+                                    Set<Mutation> suppressedMutations,
+                                    String flag) {
+
+    final int exitCode
+    final Set<Mutation> survivedAndUnsuppressedMutations =
+            setDifference(survivedMutations, suppressedMutations)
+    final Set<Mutation> unnecessarySuppressions =
+            setDifference(suppressedMutations, survivedMutations)
```

Comment:
```suggestion
    final Set<Mutation> survivedAndUnsuppressedMutations =
            setDifference(survivedMutations, suppressedMutations)
    final Set<Mutation> unnecessarySuppressions =
            setDifference(suppressedMutations, survivedMutations)
      
    final int exitCode

```

Reply:
`survivedAndUnsuppressedMutations` -> `survivingUnsuppressedMutations`

`unnecessarySuppressions` -> `extraSuppressions` (we do not yet know if they are unnecessary)

Reply:
I would prefer `surviving` to `survived` in all cases in this file.